addons/udes_stock_refactoring/tests/test_splitting.py

```diff
@@ -0,0 +1,649 @@
+        )
+
+    def _assert_move_line_fields(self, move_line, product, product_qty, package):
+        """
+        Assert that the supplied move line recordset has only 1 record.
+        Assert that the move line has the expected product, product quantity,
+        package and result package
+        """
+        self.assertEqual(len(move_line), 1)
+        self.assertEqual(move_line.product_id, product)
+        self.assertEqual(move_line.product_qty, product_qty)
+        self.assertEqual(move_line.package_id, package)
+        self.assertEqual(move_line.result_package_id, package)
+
+    def _assert_picking_fields(
+        self,
+        picking,
+        state=None,
+        origin=None,
+        partner=None,
+        group_name=None,
+        location_id=None,
+        location_dest_id=None,
+        date_done=None,
+    ):
+        """
+        Assert that the supplied picking recordset has only 1 record.
+        Assert that the state, origin, partner, group name, location,
+        destination location and date done match the expected values 
+        if supplied.
+        """
+        self.assertEqual(len(picking), 1)
+
+        if state:
+            self.assertEqual(picking.state, state)
+        if origin:
+            self.assertEqual(picking.origin, origin)
+        if partner:
+            self.assertEqual(picking.partner_id, partner)
+        if group_name:
+            self.assertEqual(picking.group_id.name, group_name)
+        if location_id:
+            self.assertEqual(picking.location_id, location_id)
+        if location_dest_id:
+            self.assertEqual(picking.location_dest_id, location_dest_id)
+        if date_done:
+            self.assertEqual(picking.date_done, date_done)
+
+
+class TestAssignSplitting(TestSplittingBase):
+    def setUp(self):
+        """
+        Set post assign action for test pick type
+        """
+        # group by package post assign
+        self.picking_type_pick.write(
+            {
+                "u_post_assign_action": "group_by_move_line_key",
+                "u_move_line_key_format": "{package_id.name}",
+            }
+        )
+
+        super(TestAssignSplitting, self).setUp()
+
+    def test_reserve_1_pallet_diff_prods_splits_1_pick_per_pallet(self):
+        """
+        With 1 pallet each for 2 different products.
+        Reserve pick (action_assign).
+        Pick should be split into 2 picks 1 for each pallet, 
+        reusing the original pick.
+        """
+        apple_pallet = self.apple_pallet
+        banana_pallet = self.banana_pallet
+
+        products_info = [{"product": self.apple, "qty": 10}, {"product": self.banana, "qty": 10}]
+        moves = self.create_move(self.picking, products_info)
+
+        apple_move = moves.filtered(lambda m: m.product_id == self.apple)
+        banana_move = moves - apple_move
+
+        self.picking.action_assign()
+
+        apple_pick = apple_move.picking_id
```

Comment:
We should check that they are different. Otherwise all the following assert could still pass without moving one of the moves into a new picking.

Reply:
Have added a comment explaining how `_assert_picking_fields` will cover this as both `apple_pick` and `banana_pick` are checked with different `group_name` values (as a picking can only have one group).